### PR TITLE
gcp - gce node groups resource initial setup and docs

### PR DESCRIPTION
**Pull request by alexkarpitski**
Added ``query`` and ``get`` methods, ``delete``, ``increase-size``, and ``decrease-size`` actions support for GCE Node Groups as well as a usage example.

As a use case, Custodian can delete Node Groups that match  a filter (have too many nodes, etc).

### CODE REVIEW

File: docs/source/gcp/examples/gce-node-group.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Compute Engine - Delete Node Groups with Invalid Number of Nodes
+================================================================
+Custodian can delete Node Groups that have too many nodes.
+
+In the example below, the policy checks if there are Node Groups whose ``size`` is greater than ``2``.
+
+.. code-block:: yaml
+    policies:
```
Comment (evhenyshaliov) [inline docs/source/gcp/examples/gce-node-group.rst]: Add extra empty row. Use tox -e docs for a check

Comment (alexkarpitski) [reply docs/source/gcp/examples/gce-node-group.rst]: Done, thanks.

**Comment by jondavidi** [comment]
@brentclements, would appreciate a second pair of eyes on this

**Comment by jondavidi** [comment]
After some further discussion, we'd like to put this PR to side for right now because I think there's some more thought that needs to go into modeling how we delete nodes from a node group, because there's concern over interfering with an application if we're deleting random nodes.

This is good work, and we want to keep it in the repository for revisiting later on, but for the remainder of this project we will just put this resource to the side for now. Thanks for the good work on this.

**Comment by jondavidi** [comment]
@aleonidex Please see the above comment for an update on this resource.

**Pull request closed without merging**